File: idealingua-v1/idealingua-v1-runtime-rpc-typescript/src/main/resources/runtime/typescript/irt/transport/auth/auth.token.ts
import { AuthMethod } from './auth.method';

export class AuthToken extends AuthMethod{
    public token: string;
    constructor(token: string) {
        super();
        this.token = token;
    }

    public fromValue(value: string): Error {
        if (value.indexOf('Bearer ') !== 0) {
            return new Error('token authorization must start with Bearer, got ' + value)
        }
        this.token = value.substr(7);
        return undefined;
    }

    public toValue(): string {
        return 'Bearer ' + this.token;
    }
}

File: idealingua-v1/idealingua-v1-runtime-rpc-csharp/src/main/resources/runtime/csharp/IRT/Transport/Authorization/AuthToken.cs

namespace IRT.Transport.Authorization {
    class AuthToken: AuthMethod {
        public string Token;

        public AuthToken(string token = null) {
            Token = token;
        }

        public override bool FromValue(string value) {
            return false;
        }

        public override string ToValue() {
            return "Bearer " + Token;
        }
    }
}